test/TestUtilities/Test.Utility/Signing/TrustedTestCert.cs

```diff
@@ -54,18 +63,49 @@ public class TrustedTestCert<T> : IDisposable
                 maximumValidityPeriod = TimeSpan.FromHours(2);
             }
 
-#if IS_DESKTOP
             if (TrustedCert.NotAfter - TrustedCert.NotBefore > maximumValidityPeriod.Value)
             {
                 throw new InvalidOperationException($"The certificate used is valid for more than {maximumValidityPeriod}.");
             }
-#endif
-            StoreName = storeName;
-            StoreLocation = storeLocation;
+
+            StoreName = GetPlatformSpecificStoreName(storeName, dir);
+            StoreLocation = GetPlatformSpecificStoreLocation(storeLocation);
             AddCertificateToStore();
+
             ExportCrl();
         }
 
+        private StoreLocation GetPlatformSpecificStoreLocation(StoreLocation requestedLocation)
+        {
+            if (requestedLocation == StoreLocation.LocalMachine &&
+                (RuntimeEnvironmentHelper.IsMacOSX || RuntimeEnvironmentHelper.IsLinux))
+            {
+                return StoreLocation.CurrentUser;
+            }
+
+            return requestedLocation;
+        }
+
+        private StoreName GetPlatformSpecificStoreName(StoreName requestedStoreName, TestDirectory dir)
+        {
+            if (RuntimeEnvironmentHelper.IsMacOSX)
+            {
+                if (requestedStoreName == StoreName.Root || requestedStoreName == StoreName.CertificateAuthority)
+                {
+                    TrustCertInMac(dir);
+                }
+
+                return StoreName.My;
```

Comment:
MacOS can only Read/Write to `My` but this doesn't make the cert trusted. An explicit command has to be run for that.